src/client/state.ml

```diff
@@ -231,6 +230,12 @@ let mk_unary_handler (self : t) ?buf_pool ~timer ~oc ~encoding ~timeout_s rpc :
 
 let default_timeout_s_ : float = 30.
 
```

Comment:
5 lines up from here: under heavy load it could be possible for the timeout handler to fire while `send_request_` is still in progress.